pandas/io/formats/format.py

```diff
@@ -451,9 +453,27 @@ def get_adjustment() -> TextAdjustment:
 class TableFormatter:
 
     show_dimensions: Union[bool, str]
-    is_truncated: bool
-    formatters: FormattersType
-    columns: Index
+
+    @property
+    def is_truncated(self) -> bool:
+        self._is_truncated: bool
```

Comment:
what is this line for?

Reply:
This is necessary to make mypy happy. For some reason it thinks that there is a violation of Liskov substitution principle even if is_truncated property (the same name as the attribute in superclass) in the subclass has the same return type. Therefore, I had to create the same property signature in the superclass.

Reply:
could you put `_is_truncated: bool` at the class level just before `__init__`?

Reply:
This is certainly possible. But considering questionable use of property setters (like discussed with @simonjayhawkins and which you imply in your other comment), probably I will just create initialization methods for each attribute where required. Therefore, there will be no need in the constructs you refer to.